File: test-publish.js
const grpc = require('@grpc/grpc-js');
const protoLoader = require('@grpc/proto-loader');

// Load gRPC protobuf definitions
const PROTO_PATH = './ai_network_dag.proto'; // Update with your actual proto file path
const packageDefinition = protoLoader.loadSync(PROTO_PATH, {
    keepCase: true,
    longs: String,
    enums: String,
    defaults: true,
    oneofs: true
});
const ainProto = grpc.loadPackageDefinition(packageDefinition).ain;

// Create a gRPC client
const client = new ainProto.AINetworkMerkleDAG('localhost:50051', grpc.credentials.createInsecure());

// Publish 요청
function testPublish() {
    const topic = 'topic';
    const cid = 'testCid';

    client.publish({ topic, cid }, (error, response) => {
        if (!error) {
            console.log('Publish response:', response);
        } else {
            console.error('Publish error:', error);
        }
    });
}

// Subscribe 요청
function testSubscribe() {
    const topic = 'topic';
    const worker = 'testWorker';

    const stream = client.subscribe({ topic, worker });

    stream.on('data', (response) => {
        console.log('Received message:', response);
    });

    stream.on('error', (error) => {
        console.error('Subscribe error:', error);
    });

    stream.on('end', () => {
        console.log('Subscription ended');
    });
}

// Run both functions
testSubscribe();
setTimeout(testPublish, 1000); // Wait 1 second before publishing

File: test-publish.py
import json
from AINetworkDagSdk import AINetworkDagSdk

def publish_message():
    # Instantiate the client
    client = AINetworkDagSdk("localhost:50051")

    # Function to publish a message
    def publish(topic, message):
        try:
            client.publish(topic, json.dumps(message))
            # print(f"Publish response: {response}")
        except Exception as e:
            print(f"Error publishing message: {e}")

    try:
        # Publish a test message
        test_topic = "topic"
        test_message = {"key": "value"}
        publish(test_topic, test_message)

    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    publish_message()
